sixtyfps_runtime/corelib/window.rs

```diff
@@ -103,6 +105,23 @@ impl crate::properties::PropertyChangeHandler for WindowRedrawTracker {
     }
 }
 
+/// This enum describes the different ways a popup can be rendered by the back-end.
+pub enum PopupWindowLocation {
+    /// The popup is rendered in its own top-level window that is know to the windowing system.
+    TopLevel(Rc<Window>),
+    /// The popup is rendered as an embedded child window at the given position.
+    ChildWindow(Point),
```

Comment:
I'm a bit torn there.
Even if the backend don't render as a separate window (which the GL backend eventually should do anyway), should there not be, conceptually, a different Rc<Window> for the popup anyway?

Reply:
In torn, too. It’s tempting but otoh it means there will be a circular dependency. So if the application code decides to drop its ref to the window while a pop up is active, then I think the window will leak?